### ISSUE TITLE
Using `six` from homebrew

### ISSUE THREAD

**Issue by Fenyutanchan** [davidchall/homebrew-hep#297]
### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I am trying to install the package of "madgraph5_amcatnlo" via command `brew install madgraph5_amcatnlo`.
However, I have installed the package of `six` at homebrew via command `brew install six`.
Therefore, it shows some problems with symbol links.

### What happened (include all command output)?

```shell
$ brew install madgraph5_amcatnlo 
```
```
==> Fetching dependencies for davidchall/hep/madgraph5_amcatnlo: eigen, cgal and fastjet
==> Fetching eigen
==> Downloading https://ghcr.io/v2/homebrew/core/eigen/manifests/3.4.0_1
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/eigen/blobs/sha256:211fd7f1d58b383e3d64335c08a376a7d8433007ce61410ead0320df34b6f4bd
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:211fd7f1d58b383e3d64335c08a376a7d8433007ce61410ead0320df34b6f4bd?se=2023-01
######################################################################## 100.0%
==> Fetching cgal
==> Downloading https://ghcr.io/v2/homebrew/core/cgal/manifests/5.5.1
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/cgal/blobs/sha256:897f5c47b844d9497988a93992966d7927917e86a12ae91a1f217480bac7727e
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:897f5c47b844d9497988a93992966d7927917e86a12ae91a1f217480bac7727e?se=2023-01
######################################################################## 100.0%
==> Fetching davidchall/hep/fastjet
==> Downloading https://fastjet.fr/repo/fastjet-3.4.0.tar.gz
######################################################################## 100.0%
==> Fetching davidchall/hep/madgraph5_amcatnlo
==> Downloading https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz
######################################################################## 100.0%
==> Downloading https://launchpad.net/mg5amcnlo/3.0/3.4.x/+download/MG5_aMC_v3.4.2.tar.gz
==> Downloading from https://launchpadlibrarian.net/647058002/MG5_aMC_v3.4.2.tar.gz
######################################################################## 100.0%
==> Installing madgraph5_amcatnlo from davidchall/hep
==> Installing dependencies for davidchall/hep/madgraph5_amcatnlo: eigen, cgal and fastjet
==> Installing davidchall/hep/madgraph5_amcatnlo dependency: eigen
==> Pouring eigen--3.4.0_1.all.bottle.tar.gz
🍺  /opt/homebrew/Cellar/eigen/3.4.0_1: 546 files, 8.4MB
==> Installing davidchall/hep/madgraph5_amcatnlo dependency: cgal
==> Pouring cgal--5.5.1.arm64_ventura.bottle.tar.gz
🍺  /opt/homebrew/Cellar/cgal/5.5.1: 3,681 files, 38.4MB
==> Installing davidchall/hep/madgraph5_amcatnlo dependency: fastjet
==> ./configure --prefix=/opt/homebrew/Cellar/fastjet/3.4.0_1 --enable-pyext --enable-allcxxplugins --with-cgal=/opt/homebrew/opt/cgal
==> make
==> make install
🍺  /opt/homebrew/Cellar/fastjet/3.4.0_1: 250 files, 13MB, built in 48 seconds
==> Installing davidchall/hep/madgraph5_amcatnlo
==> python3.10 -c import setuptools... --no-user-cfg install --prefix=/opt/homebrew/Cellar/madgraph5_amcatnlo/3.4.2 --install-scripts=/opt/homebrew/Cellar/madgr
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /opt/homebrew
Could not symlink lib/python3.10/site-packages/six-1.16.0-py3.10.egg-info/PKG-INFO
Target /opt/homebrew/lib/python3.10/site-packages/six-1.16.0-py3.10.egg-info/PKG-INFO
is a symlink belonging to six. You can unlink it:
  brew unlink six

To force the link and overwrite all conflicting files:
  brew link --overwrite madgraph5_amcatnlo

To list all files that would be deleted:
  brew link --overwrite --dry-run madgraph5_amcatnlo

Possible conflicting files are:
/opt/homebrew/lib/python3.10/site-packages/six-1.16.0-py3.10.egg-info/PKG-INFO -> /opt/homebrew/Cellar/six/1.16.0_3/lib/python3.10/site-packages/six-1.16.0-py3.10.egg-info/PKG-INFO
/opt/homebrew/lib/python3.10/site-packages/six-1.16.0-py3.10.egg-info/SOURCES.txt -> /opt/homebrew/Cellar/six/1.16.0_3/lib/python3.10/site-packages/six-1.16.0-py3.10.egg-info/SOURCES.txt
/opt/homebrew/lib/python3.10/site-packages/six-1.16.0-py3.10.egg-info/dependency_links.txt -> /opt/homebrew/Cellar/six/1.16.0_3/lib/python3.10/site-packages/six-1.16.0-py3.10.egg-info/dependency_links.txt
/opt/homebrew/lib/python3.10/site-packages/six-1.16.0-py3.10.egg-info/top_level.txt -> /opt/homebrew/Cellar/six/1.16.0_3/lib/python3.10/site-packages/six-1.16.0-py3.10.egg-info/top_level.txt
/opt/homebrew/lib/python3.10/site-packages/six.py -> /opt/homebrew/Cellar/six/1.16.0_3/lib/python3.10/site-packages/six.py
==> Summary
🍺  /opt/homebrew/Cellar/madgraph5_amcatnlo/3.4.2: 4,559 files, 117.8MB, built in 4 seconds
==> Running `brew cleanup madgraph5_amcatnlo`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
```

### What did you expect to happen?

I think that the package of `six` should be installed from homebrew directly instead of manually installation with the code of
```ruby
  resource "six" do
    url "https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz"
    sha256 "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926"
  end
```
and
```ruby
    resource("six").stage do
      system python, *Language::Python.setup_install_args(prefix, python)
    end
```

Therefore, I try to modify the file of `madgraph5_amcatnlo.rb` with adding the code of
```ruby
  depends_on "six"
```
and commenting the code of
```ruby
  # resource "six" do
  #   url "https://files.pythonhosted.org/packages/71/39/171f1c67cd00715f190ba0b100d606d440a28c93c7714febeca8b79af85e/six-1.16.0.tar.gz"
  #   sha256 "1e61c37477a1626458e36f7b1d82aa5c9b094fa4802892072e49de9c60c4c926"
  # end

...

    # resource("six").stage do
    #   system python, *Language::Python.setup_install_args(prefix, python)
    # end
```

Then I reinstall the package of `madgraph5_amcatnlo` and it does work and passes the test.
If there is no any other problems, I suggested this code modification to use the package of `six` from homebrew.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
$ brew install madgraph5_amcatnlo.rb
```

### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew doctor
```
```
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  madgraph5_amcatnlo
```

**Comment by davidchall** [comment]
Thanks for raising this issue, @Fenyutanchan. I wasn't aware that Homebrew provides a `six` formula - it looks like this was a fairly recent addition (https://github.com/Homebrew/homebrew-core/pull/74909). This is definitely the right way to do it, so I'll accept your PR. 👍

**Comment by davidchall** [comment]
Closed by #298
